src/layout/Payment/config.ts

```diff
@@ -0,0 +1,54 @@
+import { CG } from 'src/codegen/CG';
+import { ExprVal } from 'src/features/expressions/types';
+import { CompCategory } from 'src/layout/common';
+
+export const Config = new CG.component({
+  category: CompCategory.Presentation,
+  rendersWithLabel: false,
+  capabilities: {
+    renderInTable: false,
+    renderInButtonGroup: false,
+    renderInAccordion: false,
+    renderInAccordionGroup: false,
+  },
+})
+  .addProperty(
+    new CG.prop(
+      'renderAsSummary',
+      new CG.expr(ExprVal.Boolean)
+        .optional({ default: false })
+        .setTitle('Render as summary')
+        .setDescription(
+          'Boolean value or expression indicating if the component should be rendered as a summary. Defaults to false.',
+        ),
+    ),
+  )
```

Comment:
Don't define this yourself, extend `SummarizableComponentProps` instead (you can do that by calling `Config.inner.extends(CG.common('SummarizableComponentProps'))`).

In the branch I'm working on I've changed that type, because it claims that `renderAsSummary` can be an expression - but support for that was never been implemented, and nobody has asked about it, so apparently nobody ever needed for it to support expressions anyway.